Python/bltinmodule.c

```diff
@@ -2426,7 +2426,8 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
         }
         s = PyOS_Readline(stdin, stdout, promptstr);
         if (s == NULL) {
-            PyErr_CheckSignals();
+            if (!PyErr_Occurred())
+                PyErr_CheckSignals();
             if (!PyErr_Occurred())
                 PyErr_SetNone(PyExc_KeyboardInterrupt);
```

Comment:
I don't like `if` clauses that have the same condition right next to each other. Let's refactor to something like this:

```suggestion
            if (!PyErr_Occurred()) {
                if (PyErr_CheckSignals() == 0) {
                    PyErr_SetNone(PyExc_KeyboardInterrupt);
                }
            }
```

Reply:
Yeah, it looks a little goofy. I'm happy to take your version, but looking at it again, what about:

```C
if (!PyErr_Occurred() && !PyErr_CheckSignals()) {
    PyErr_SetNone(PyExc_KeyboardInterrupt);
}
```